src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/HotReload/ProjectHotReloadSessionManager.cs

```diff
@@ -449,24 +485,25 @@ async Task ApplyHotReloadUpdateInternalAsync()
             }
         }
 
-        private class HotReloadState : IProjectHotReloadSessionCallback
+        internal class HotReloadState : IProjectHotReloadSessionCallback
         {
-            private readonly ProjectHotReloadSessionManager _sessionManager;
-
             public Process? Process { get; set; }
             public IProjectHotReloadSession? Session { get; set; }
 
-            // TODO: Support restarting the session.
+            public ProjectHotReloadSessionManager SessionManager { get; }
+
             public bool SupportsRestart => false;
```

Comment:
Should this return true? Earlier, we check if `_callback is HotReloadState` and return `true` if so.

If we changed this to return true, then perhaps we could just delegate directly in that code rather than doing a type check.

Reply:
Resolved in #[HotReloadState](https://github.com/dotnet/project-system/pull/9555#discussion_r1838970838)